pandas/tests/dtypes/test_dtypes.py

```diff
@@ -742,3 +744,31 @@ def test_categorical_categories(self):
         tm.assert_index_equal(c1.categories, pd.Index(['a', 'b']))
         c1 = CategoricalDtype(CategoricalIndex(['a', 'b']))
         tm.assert_index_equal(c1.categories, pd.Index(['a', 'b']))
+
+
+class DummyArray(ExtensionArray):
+    pass
+
+
+class DummyDtype(ExtensionDtype):
+    pass
+
+
+class TestExtensionArrayDtype(object):
+
+    @pytest.mark.parametrize('values', [
+        pd.Categorical([]),
+        pd.Categorical([]).dtype,
+        pd.Series(pd.Categorical([])),
+        DummyDtype(),
+        DummyArray(),
```

Comment:
can you test all of the exposed methods / properties

Reply:
The only thing I could test on the dummy ones is `kind`, which returns just 'O' and `names`, which I'm hopeful we can remove.

All the other methods are abstract or rely on an abstract method.

I do have real classes that implement the interface, but waiting for the interface to settle a bit before rewriting those classes and the tests.